### PR TITLE
Adding new EntitySwitch for differing views depending on ResourceType

### PR DESCRIPTION
**Pull request by dlaird-ovo**
## Added new Entity Switch

I have added a `isResourceType` EntitySwitch.  This is so that in the EntityPage I can use this construct:
```ts
    <EntityLayout.Route if={isResourceType('kubernetes-cluster')} path="/kubernetes" title="Kubernetes">
      ...
    </EntityLayout.Route>
```

This allows me to show different widgets on the Resources pages - which can be of various types:
- S3 Buckets
- kubernetes-clusters
- databases
- etc...

#### :heavy_check_mark: Checklist

- [X] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [X] Added or updated documentation
- [X] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [X] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: plugins/catalog/src/components/EntitySwitch/conditions.test.ts [hunk 1]
```diff
@@ -59,6 +78,26 @@ const bNamespace: Entity = {
   spec: { type: 'service' },
 };
 
+describe('isResourceType', () => {
+  it('should false on non component kinds', () => {
+    const checkEntity = isResourceType('service');
+
+    expect(checkEntity(notComponent)).not.toBeTruthy();
```
Comment (freben) [inline plugins/catalog/src/components/EntitySwitch/conditions.test.ts:85]: notComponent wasn't really a variable i expected here :) this reads odd overall

**Review by freben — COMMENTED** [review]
Nice! Just a small tweak to the test

File: plugins/catalog/src/components/EntitySwitch/conditions.test.ts [hunk 2]
```diff
@@ -59,6 +78,26 @@ const bNamespace: Entity = {
   spec: { type: 'service' },
 };
 
+describe('isResourceType', () => {
+  it('should false on non component kinds', () => {
```
[the remark this answers is not in the document]
Comment (dlaird-ovo) [reply plugins/catalog/src/components/EntitySwitch/conditions.test.ts]: Thankyou for feedback - addressed in new commit

File: plugins/catalog/src/components/EntitySwitch/conditions.test.ts (hunk 1 above)
Comment (dlaird-ovo) [reply plugins/catalog/src/components/EntitySwitch/conditions.test.ts]: Thankyou for feedback - addressed in new commit

File: plugins/catalog/src/components/EntitySwitch/conditions.test.ts [hunk 3]
```diff
@@ -15,7 +15,26 @@
...
+import {
+  isComponentType,
+  isKind,
+  isNamespace,
+  isResourceType,
+} from './conditions';
+
+const kubernetesClusterResource: Entity = {
+  apiVersion: '',
+  kind: 'resource',
```
[the remark this answers is not in the document]
Comment (dlaird-ovo) [reply plugins/catalog/src/components/EntitySwitch/conditions.test.ts]: 👍

**Review by freben — APPROVED** [review]

**Pull request merged**